packages/format-library/src/link/inline.js

```diff
@@ -178,17 +175,16 @@ function InlineLinkUI( {
 				newValue = concat( valBefore, newValAfter );
 			}
 
-			newValue.start = newValue.end;
```

Comment:
This line was added by @ellatrix in https://github.com/WordPress/gutenberg/pull/17126.

I proposed we revert it. Why? The new UX that this PR is aiming to implement for links is to leave the Link UI open when first submitting a link. This provides an improvement because 

- users can validate they added the correct link
- users can modify the link settings after they added it

(further rationale is available on the linked Issue).

Unfortunately, by collapsing the selection and placing it _outside_ the link format we just created/edited we make it impossible for our APIs to determine how to apply further updates to the link format because it cannot "find" the format (because the format is no longer technically within the selection range).

This was what was causing [this bug](https://github.com/WordPress/gutenberg/pull/57726#issuecomment-1888860129) with `Opens in new tab` updating not being persisted.

When assessing whether or not we should drop this line, it's worth reviewing @WunderBart's screenshots from his approval in 2020 which show what UX this aims to achieve - namely that once a link is created the cursor moves outside the link to allow you to continue typing.

Reply:
I think this change is good because linking is a format and all other formats keep the selection. The small speed gain that allows typing right after linking is not worth the trade off of not seeing confirmation of your applied format.

Reply:
I think it makes sense to remove the change if we want to keep the Link UI open as the two things are connected - when we hide the UI we want to move the cursor to the end to enable you to carry on typing. However if we keep the UI open then we should leave the cursor where it is.

Reply:
OK let's consider this resolved.